### PR TITLE
Remove gradient checkpointing option from various training scripts 

### PR DESCRIPTION
**Pull request by qgallouedec**
`gradient_checkpointing` is true by default, there is no need to have it everywhere

### CODE REVIEW

File: tests/test_grpo_trainer.py [hunk 1]
```diff
@@ -414,55 +414,6 @@ def test_training_peft_model(self):
             elif "base_layer" not in n and "ref" not in n:  # and the peft params to be different (except base and ref)
                 assert not torch.allclose(param, new_param), f"Parameter {n} has not changed."
 
-    @require_peft
-    def test_training_peft_with_gradient_checkpointing(self):
-        """Test that training works with PEFT and gradient checkpointing enabled."""
```
Comment (albertvillanova) [inline tests/test_grpo_trainer.py:419]: Shouldn't we keep this test? It specifically validates that gradient checkpointing works correctly with PEFT models. Even if gradient checkpointing is enabled by default, this test serves to verify that the integration doesn't break. Consider keeping this test but renaming it to reflect that it tests the default behavior (e.g., `test_training_peft_with_default_gradient_checkpointing`).

File: tests/test_reward_trainer.py [hunk 2]
```diff
@@ -446,42 +446,6 @@ def test_train_moe_with_peft_config_and_gradient_checkpointing(self):
             elif "base_layer" not in n:  # We expect the peft parameters to be different (except for the base layer)
                 assert not torch.allclose(param, new_param), f"Parameter {n} has not changed"
 
-    @require_peft
-    def test_train_with_peft_model_and_gradient_checkpointing(self):
-        # Get the base model parameter names
```
Comment (albertvillanova) [inline tests/test_reward_trainer.py:451]: The same as above: it validates a specific combination of PEFT + gradient checkpointing. The removal reduces test coverage for this important use case. Consider retaining but updating the docstring to clarify it tests default behavior.

File: tests/test_dpo_trainer.py [hunk 3]
```diff
@@ -1515,7 +1515,6 @@ def test_dpo_peft_model(self, model_id, loss_type, pre_compute_logits, gradient_
             fp16=True,
             logging_strategy="no",
             report_to="none",
-            gradient_checkpointing=True,
```
Comment (albertvillanova) [inline tests/test_dpo_trainer.py:1518]: Maybe adding an explanatory comment about passing `gradient_checkpointing_kwargs` below because gradient_checkpointing is True by default?

File: tests/test_grpo_trainer.py (hunk 1 above)
Comment (qgallouedec) [reply tests/test_grpo_trainer.py:419]: Okay, I understand the reasoning. Let's keep one test that explicitly tests the GC, with a clear comment explaining why we have it.

https://github.com/huggingface/trl/pull/4905/commits/e41654b40fc1493ca888ab6b42307ab218df09cc

File: tests/test_dpo_trainer.py (hunk 3 above)
Comment (qgallouedec) [reply tests/test_dpo_trainer.py:1518]: a reverted this change and added a comment in https://github.com/huggingface/trl/pull/4905/commits/e41654b40fc1493ca888ab6b42307ab218df09cc

File: tests/test_grpo_trainer.py (hunk 1 above)
Comment (qgallouedec) [reply tests/test_grpo_trainer.py:419]: oops I missed grpo

Comment (qgallouedec) [reply tests/test_grpo_trainer.py:419]: https://github.com/huggingface/trl/pull/4905/commits/77e4e48f0b1202f0f8e43eea6929110567d2b510
